app/main/routes.py

```diff
@@ -49,55 +49,56 @@ def upload():
 
     if request.method == "POST":
         excel_file = request.files.get("ingest_spreadsheet")
-        error = check_file(excel_file)
-        if error:
-            current_app.logger.info(
-                PRE_VALIDATION_ERROR_LOG.format(error=error) if Config.ENABLE_VALIDATION_LOGGING else PRE_VALIDATION_LOG
-            )
-            return render_template(
-                "upload.html",
-                pre_error=[error],
-                days_to_deadline=calculate_days_to_deadline(),
-                reporting_period=Config.REPORTING_PERIOD,
-                fund=Config.FUND_NAME,
+        if not (pre_errors := check_file(excel_file)):
+            pre_errors, validation_errors, metadata = post_ingest(
+                excel_file,
+                {"reporting_round": 4, "auth": json.dumps(g.auth.get_auth_dict()), "do_load": is_load_enabled()},
```

Comment:
A more minor point to my points in slack but line 52 + reassigning `pre_errors` is a bit cognitively overloading for me.

We also potentially get to the next `if`/`else` block without instantiating validation errors and metadata which isn't clean and could open us up to errors.

I think something like this may be clearer:
```
        if file_errors := check_file(excel_file):
            pre_errors, validation_errors, metadata = file_errors, None, None
        else:
            pre_errors, validation_errors, metadata = post_ingest(
                excel_file,
                {"reporting_round": 4, "auth": json.dumps(g.auth.get_auth_dict()), "do_load": is_load_enabled()},
            )
```